src/Seq.Input.AzureEventHub/ClefEventProcessor.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.Azure.EventHubs;
+using Microsoft.Azure.EventHubs.Processor;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Seq.Input.AzureEventHub
+{
+    public class ClefEventProcessor : IEventProcessor
+    {
+        private readonly object processLock = new object();
```

Comment:
If the event hubs client creates multiple `ClefEventProcessor` instances (e.g. one per partition) then they will each end up with their own `processLock` and won't synchronize; perhaps passing the lock in through the constructor, from `AzureEventHubListener` would be a quick way to achieve synchronization?

(Having the processor call out to a third object that encapsulates both the lock and the protected `inputWriter`, e.g. your own `SynchronizedInputWriter`, would be more robust against accidental creation of deadlocks through future code changes.)

Reply:
Good idea. I implemented a helper class.